code/modules/custom_ka/core.dm

```diff
@@ -234,19 +234,43 @@
 		installed_upgrade_chip.on_fire(src)
 	if(installed_barrel)
 		installed_barrel.on_fire(src)
-
-	if(ispath(installed_barrel.projectile_type, /obj/item/projectile/kinetic))
-		var/obj/item/projectile/kinetic/shot_projectile = new installed_barrel.projectile_type(get_turf(src))
-		shot_projectile.damage = damage_increase
-		shot_projectile.range = range_increase
-		shot_projectile.aoe = max(1, aoe_increase)
-		shot_projectile.base_damage = damage_increase
-		return shot_projectile
-	if(ispath(installed_barrel.projectile_type, /obj/item/projectile/beam))
-		var/obj/item/projectile/beam/shot_projectile = new installed_barrel.projectile_type(get_turf(src))
-		shot_projectile.damage = damage_increase
-		shot_projectile.range = range_increase
-		return shot_projectile
+		
+	//Get turf and see if in atmos or not
+	var/turf/T = get_turf(src)
+	
+	if(T)
+		var/datum/gas_mixture/environment = T.return_air()
+		var/pressure = (environment)? environment.return_pressure() : 0
+		//If pressure is greater than about 40 kPA
+		if(pressure > ONE_ATMOSPHERE*0.4)
+			if(ispath(installed_barrel.projectile_type, /obj/item/projectile/kinetic))
+				var/obj/item/projectile/kinetic/shot_projectile = new installed_barrel.projectile_type(get_turf(src))
+				shot_projectile.damage = damage_increase
+				shot_projectile.range = range_increase
```

Comment:
this is a lot of copypaste, do this on a generic projectile typecast, then edit the  other types as needed